src/platform_impl/linux/wayland/event_loop.rs

```diff
@@ -714,8 +714,43 @@ impl<T> EventLoop<T> {
         window_target.store.lock().unwrap().for_each(|window| {
             let window_id =
                 crate::window::WindowId(crate::platform_impl::WindowId::Wayland(window.wid));
-            if let Some(frame) = window.frame {
-                if let Some((w, h)) = window.newsize {
+
+            // Update window logical .size field (for callbacks using .inner_size)
+            let (old_size, mut logical_size) = {
+                let mut window_size = window.size.lock().unwrap();
+                let old_size = *window_size;
+                *window_size = window.new_size.unwrap_or(old_size);
+                (old_size, *window_size)
+            };
+
+            if let Some(scale_factor) = window.new_scale_factor {
+                // Update cursor scale factor
+                self.cursor_manager
```

Comment:
Have you checked that lock will be released?

Reply:
Is there a way the guard could escape ?

Reply:
I mean the problem is knowing exactly the time lock will be dropped, I'm just ensuring that it'll be drop in the right time. I feel like lock will be acquired until the end of scope, which is after `callback`? At least I seen issues similar to that already.

Reply:
The update is scoped in a block so that the mutex is unlocked there when the mutex guard drops out of the block scope.

Reply:
yeah, it's fine. agree.